CONTRIBUTING.md

```diff
@@ -0,0 +1,198 @@
+
+### Core Features
+
+#### Collecting Logs
+
+The extension receives each activity log in the form of an `object` from activityLog API. The activityLog API schema can be found [here](https://searchfox.org/mozilla-central/source/toolkit/components/extensions/schemas/activity_log.json). The background page (backend part) subscribes the `browser.activityLog.onExtensionActivity` API event listener, which gets triggered with an activity log object passed as a parameter everytime an activity is found from any monitored extension. While monitoring other extensions, any newly installed extension is being monitored automatically.
+
+- ##### Live Logging
+
+  The real-time logs are collected in the background and sent to Activity Log page while it is opened. The extension uses `runtime.sendMessage` API to communicate between background and Activity Log page. The [`sendLogs`](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-monitor.js#L25-L33) method is responsible for sending logs to Activity Log page. The Activity Log page listens for logs via [`runtime.onMessage`](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-activitylog.js#L253-L265) event and render those in the [`log-view`](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/web-component/log-view/).
+
+- ##### Loading / saving logs
+
+  It can save all the collected logs that are collected in the background to a JSON file. To save logs, the `saveLogs` instruction from [`save-load.js`](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/save-load.js#L22-L27) is sent to the background where the save operation occurs.
+
+  The saved logs can be viewed by loading the JSON file. It will read the logs and create a new tab. The tab id along with the loaded logs are saved in the Background since we want to keep the loaded logs accessible on the same tab (where the logs were initially loaded) and delete the logs once that tab is closed, we temporarily store the [loaded logs by the tab id](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-monitor.js#L6) in Background.
+
+  The Activity Log page initializes in the new tab by fetching the loaded logs from the Background with it's tab id and render them in a [`log-view`](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/web-component/log-view/). When the tab is closed, the loaded logs along with the tab id will be removed from Background.
```

Comment:
This thing about the tabId looks out of place here in "Loading / saving logs". Remove it. It's mentioned later in filtering log entries anyway.

Reply:
Removed :-)